packages/compiler/src/typekit/kits/array.ts

```diff
@@ -29,7 +29,10 @@ defineKit<TypekitExtension>({
   array: {
     is(type) {
       return (
-        type.kind === "Model" && isArrayModelType(this.program, type) && type.properties.size === 0
+        this.type.is(type) &&
```

Comment:
```suggestion
        type.entityKind === "Type" &&
```
minor perf boost so it doesn't go through proxy I think

Reply:
That's fair - I'll update